library/kotlin/io/envoyproxy/envoymobile/StreamPrototype.kt

```diff
@@ -21,7 +22,7 @@ open class StreamPrototype(private val engine: EnvoyEngine) {
    * @param executor Executor on which to receive callback events.
    * @return The new stream.
    */
-  open fun start(executor: Executor): Stream {
+  open fun start(executor: Executor = Executors.newSingleThreadExecutor()): Stream {
```

Comment:
You know best, just wanted to raise for debate: should we provide an executor by default or have people provide one so that they _have_ to think about it?

Reply:
Soooo, the reason I did this is because the swift api is `start()` so I figured this would at least bring some visual parity....

I also did this cause i didn't want to write this for every unit test... 😛

Reply:
> I also did this cause i didn't want to write this for every unit test... 😛

I _knew_ it... hahaha